app/src/main/java/com/sunasterisk/musixmatch/data/source/local/PlaylistLocalDataSource.java

```diff
@@ -0,0 +1,203 @@
+
+    private static final String CONTENT_URI_NAME = "external";
+    private static final String CONTENT_URI_MEDIA = "content://media";
+    private static final int INCREMENT_NUMBER = 1;
+    private static final int FIRST_ELEMENT_INDEX = 0;
+    private static final int ERROR_INDEX = -1;
+    private static final int SIZE = 1;
+    private static PlaylistLocalDataSource sInstance;
+    private ContentResolver mContentResolver;
+
+    private PlaylistLocalDataSource(Context context) {
+        mContentResolver = context.getContentResolver();
+    }
+
+    public static PlaylistLocalDataSource getsInstance(Context context) {
+        if (sInstance == null) {
+            sInstance = new PlaylistLocalDataSource(context);
+        }
+        return sInstance;
+    }
+
+    @Override
+    public void getPlaylist(Callback<List<Playlist>> callback) {
+        AsyncTask<Void, Void, List<Playlist>> asyncTask = new AsyncTask<Void, Void, List<Playlist>>() {
+            @Override
+            protected List<Playlist> doInBackground(Void... voids) {
+                return getLocalPlaylist();
+            }
+
+            @Override
+            protected void onPostExecute(List<Playlist> playlists) {
+                super.onPostExecute(playlists);
+                callback.getDataSuccess(playlists);
+            }
+        };
+        asyncTask.execute();
+    }
+
+    @Override
+    public Uri addSongToPlaylist(long playlistId, Track track) {
+        if (playlistId < 0) {
+            return null;
+        }
+        // Find the greatest PLAY_ORDER in the playlist
+        int base = getLargestPos(playlistId);
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(MediaStore.Audio.Playlists.Members.PLAY_ORDER, base + INCREMENT_NUMBER);
+        contentValues.put(MediaStore.Audio.Playlists.Members.AUDIO_ID, track.getTrackId());
+        Uri uri = mContentResolver.insert(MediaStore.Audio.Playlists.Members.getContentUri(CONTENT_URI_NAME, playlistId), contentValues);
+        mContentResolver.notifyChange(Uri.parse(CONTENT_URI_MEDIA), null);
+        return uri;
+    }
+
+    public int getLargestPos(long playlistId) {
+        Uri uri = MediaStore.Audio.Playlists.Members.getContentUri(CONTENT_URI_NAME, playlistId);
+        String[] projection = new String[]{MediaStore.Audio.Playlists.Members.PLAY_ORDER};
+        Cursor cursor = mContentResolver.query(uri, projection, null, null, null);
+        int base = FIRST_ELEMENT_INDEX;
+        if (cursor.moveToLast()) {
+            base = cursor.getInt(FIRST_ELEMENT_INDEX) + INCREMENT_NUMBER;
+        }
+        cursor.close();
+        return base;
+    }
+
+    @Override
+    public long createPlaylist(String name) {
+        long id = getPlaylistId(name);
+
+        if (id == ERROR_INDEX) {
+            // We need to create a new playlist.
+            ContentValues values = new ContentValues(INCREMENT_NUMBER);
```

Comment:
```suggestion
            ContentValues values = new ContentValues();
```
INCREMENT_NUMBER = 1  dùng để tăng giá trị mỗi bước lên 1
bh giả sử mỗi lần increate a muốn tăng 10 INCREMENT_NUMBER = 10
thì em tính sửa đoạn này ntn

Reply:
với mỗi 1 biến chỉ sử dụng với 1 ý nghĩa của nó
k phải vì nó có cùng giá trị mà sử dung được